driver-core/src/main/com/mongodb/operation/CommitTransactionOperation.java

```diff
@@ -46,14 +47,15 @@
 @Deprecated
 public class CommitTransactionOperation extends TransactionOperation {
```

Comment:
I think something is missing, both here and for AbortTransactionOperation.  The specification requires that the session is unpinned from its mongos on a retryable failure of either command, _before_ the retry is attempted.  Since the retry actually happens in `CommandOperationHelper#executeRetryableCommand`, that would indicate that the unpinning also has to occur in that method, before the retry is attempted.

This is what I was talking about when I said that I think that the SessionContext has to support unpinning, since the SessionContext is all that we have available within the operation.